dm/checker/checker.go

```diff
@@ -144,6 +147,23 @@ func (c *Checker) Init(ctx context.Context) (err error) {
 	if egErr := eg.Wait(); egErr != nil {
 		return egErr
 	}
+	// check connections
+	if _, ok := c.checkingItems[config.ConnAmountChecking]; ok {
+		if len(c.stCfgs) > 0 && len(c.instances) > 0 {
+			switch c.stCfgs[0].Mode {
+			case config.ModeAll:
```

Comment:
we can left a TODO for syncer here, otherwise reviewer may get confused why the two cases are same

Reply:
20edb4b